### PR TITLE
Fix drop of completely dropped table

### PR DESCRIPTION
**Pull request by alesapin**
### Changelog category (leave one):
- Not for changelog (changelog entry is not required)

### Changelog entry (a user-readable short description of the changes that goes to CHANGELOG.md):

### CODE REVIEW

File: src/Storages/StorageReplicatedMergeTree.cpp [hunk 1]
```diff
@@ -7479,6 +7479,10 @@ void StorageReplicatedMergeTree::createTableSharedID() const
             id = zookeeper->get(zookeeper_table_id_path);
             LOG_DEBUG(log, "Shared ID on path {} concurrently created, will set ID {}", zookeeper_table_id_path, id);
         }
+        else if (code == Coordination::Error::ZNONODE)
+        {
+            LOG_DEBUG(log, "Shared ID on path {} cannot be created, table is completely dropped, will use {}", zookeeper_table_id_path, id);
+        }
```
Comment (tavplubix) [inline src/Storages/StorageReplicatedMergeTree.cpp:7485]: So `table_shared_id` will not be safe and `getTableSharedID` will remove Nil, is it ok? Of course we don't really care about dropped table, but it breaks invariant "getTableSharedID must return something meaningful"

Comment (alesapin) [reply src/Storages/StorageReplicatedMergeTree.cpp:7485]: Yes, it's ok to return random trash from this function.

Comment (tavplubix) [reply src/Storages/StorageReplicatedMergeTree.cpp:7485]: Maybe remove this branch and move `createTableSharedID` from ctor to `startup`?

**Review by tavplubix — APPROVED** [review]

**Comment by alesapin** [comment]
Failed tests are flaky. 
But failure with UB san is scarry to me:
```
Calculated checksum of the executable (136378F0C2EB58108AA3B7A357EEC13A) does not correspond to the reference checksum stored in the executable (06A610082550DFA727475EBE3BB01E18). This may indicate one of the following: - the executable was changed just after startup; - the executable was corrupted on disk due to faulty hardware; - the loaded executable was corrupted in memory due to faulty hardware; - the file was intentionally modified; - a logical error in the code.

```

**Pull request merged**